src/goose/synopsis/toolkit.py

```diff
@@ -242,3 +245,36 @@ def change_dir(self, path: str) -> str:
         self.logshell(f"cd {path}")
         system.cwd = str(patho)
         return path
+
+    @tool
+    def fetch_web_content(self, url: str) -> str:
```

Comment:
this might be worth adding `@cache` though i'm not quite sure that plays well with the named temp file

Reply:
I think temp files that is implicit?

Reply:
@michaelneale the named file is created each time this function is called so it's not cached at all

https://github.com/block/goose/blob/e19006c4a0c743ed90cbbdd51a817eb86afcbc5d/src/goose/toolkit/developer.py#L97 

if we add `@cache` we'd memorize the function call in memory so it wouldn't need to hit disk

Reply:
the LLM gets the file back, not the content, so it won't call the fetch each time as it knows in its context that it has the file with the content

Reply:
ah right, that makes sense